substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -221,39 +223,44 @@ pub mod pallet {
 		/// After the threshold is reached a new era will be forced.
 		type OffendingValidatorsThreshold: Get<Perbill>;
 
-		/// Something that provides a best-effort sorted list of voters aka electing nominators,
+		/// Something that provides a best-effort sorted list of voters (aka electing nominators),
 		/// used for NPoS election.
 		///
 		/// The changes to nominators are reported to this. Moreover, each validator's self-vote is
 		/// also reported as one independent vote.
 		///
+		/// Voters will be removed from this list when their ledger is killed or the nominator is
+		/// chilled.
+		///
 		/// To keep the load off the chain as much as possible, changes made to the staked amount
 		/// via rewards and slashes are not reported and thus need to be manually fixed by the
 		/// staker. In case of `bags-list`, this always means using `rebag` and `putInFrontOf`.
 		///
-		/// Invariant: what comes out of this list will always be a nominator.
+		/// Invariant: what comes out of this list will always be an active nominator.
 		type VoterList: SortedListProvider<Self::AccountId, Score = VoteWeight>;
 
-		/// WIP: This is a noop as of now, the actual business logic that's described below is going
-		/// to be introduced in a follow-up PR.
+		/// Something that provides a sorted list of targets (aka electable and chilled
+		/// validators), used for NPoS election.
 		///
-		/// Something that provides a best-effort sorted list of targets aka electable validators,
-		/// used for NPoS election.
+		/// The changes to the approval stake of each validator are reported to this list through
+		/// [`Self::EventListeners`]. The target(s) approval stake in the list should be updated in
```

Comment:
Honestly, if I want to be pedantic, all of this is implementation detail :D There's nothing in this code that strictly links `EventListeners` to `TargetList`.

I do think it is useful and "bug-preventive" to think at the right abstraction here. 

`VoterList` and `TargetList` are totally unknown to staking. All it knows is that these two will give me my sorted list of pre-election ovters and targets. Period. 

`EventListeners` is where staking must accurately report changes to stakers' status. 

If and if not the two are related at the runtime level, we don't know.